### DISCUSSION TITLE
Issue With the Sel Determination

### DISCUSSION THREAD

**Discussion by mcourtiel** [deepmodeling/deepmd-kit#4163] [Q&A]
Dear Developers,

I am trying to calculate the sel of a system using the dp neighbor-stat tool, but it is returning an empty sel except for one of the atom types (H).
I have a periodic system in a hexagonal unit cell. This system contains 4 atom types: "Br H Mo O," labeled respectively "0 1 2 3" in the type.raw and type_map.raw files.

I am using deepMD-kit V2.2.10, and the command line is:
dp neighbor-stat -s init_MoBr -r 6.0 -t Br H Mo O
The output is the following:
DEEPMD INFO    training data with min nbor dist: 0.4716052460030911
DEEPMD INFO    training data with max nbor size: [  0 189   0   0]
DEEPMD INFO    min_nbor_dist: 0.471605
DEEPMD INFO    max_nbor_size: [  0 189   0   0]

I’ve never had this problem before. I’ve verified that all the units for the coordinates and the box.raw are in Ångström.
I also tried increasing the cutoff, but this only resulted in increasing the number of H neighbors in the cell.
I tried using a cubic sel since it’s the first time I’m training on a hexagonal system, but there were no changes.

Maybe I am incorrectly converting the cell parameters to Cartesian coordinates:
   ABC [angstrom] 15.297 15.297 11.219
   ALPHA_BETA_GAMMA [deg] 90.0 90.0 120.0
Gives : (xx xy xz yx yy yz zx zy zz)
 15.297  -7.6485   0   0  13.2454   0   0   0   11.219
But even if something is wrong, I think that I should still have other atoms detected.

Any help is would be much appreciated!

**Comment by njzjz** [comment]
Could you provide an example?

**Comment by mcourtiel** [comment]
Yes of course, you'll find in copy a file containing all the data with which I made the "set.000" thanks to the "raw to set" script.
There are 999 frames in the set.000.

[init_MoBr.tar.gz](https://github.com/user-attachments/files/17119997/init_MoBr.tar.gz)

**Comment by njzjz** [comment]
`type_map.raw` is incorrect. The shape should be the number of elements.

**Reply by mcourtiel** [reply]
Thanks a lot, it's working perfectly now.
